qiskit/visualization/pulse/interpolation.py

```diff
@@ -41,16 +41,42 @@ def interp1d(time, samples, nop, kind='linear'):
     dt = time[1] - time[0]
 
     time += 0.5 * dt
+
     cs_ry = interpolate.interp1d(time[:-1], re_y, kind=kind, bounds_error=False)
     cs_iy = interpolate.interp1d(time[:-1], im_y, kind=kind, bounds_error=False)
 
     time_ = np.linspace(time[0], time[-1] * dt, nop)
 
     return time_, cs_ry(time_), cs_iy(time_)
 
+def step_wise(time, samples, nop, kind='nearest'):
+    """Scipy interpolation wrapper.
+​
+    Args:
+        time (ndarray): time.
+        samples (ndarray): complex pulse envelope.
+        nop (int): data points for interpolation.
+        kind (str): Scipy interpolation type. See `scipy.interpolate.interp1d` documentation
+            for more information.
+    Returns:
+        ndarray: interpolated waveform.
+    """
+
+    samples_ = np.repeat(samples, 2)
+    re_y_ = np.real(samples_)
+    im_y_ = np.imag(samples_)
+
+    time__ = np.concatenate(([time[0]], np.repeat(time[1:-1], 2), [time[-1]]))
+
+    cs_ry_ = interpolate.interp1d(time__, re_y_, bounds_error=False)
```

Comment:
you need to set `kind='nearest'` here. because `step_wise` function is dedicated to `nearest` interpolation, the parameter `nearest` should be hard-corded rather than exposed in args.